### ISSUE TITLE
trimming sequence length then collapseNoMismatch() creates error

### ISSUE THREAD

**Issue by ltipton** [benjjneb/dada2#459]
I am most likely doing this in the wrong order, but I am trying to get sequence tables with varying sequence lengths, after I have run dada2 on individual MiSeq runs and merged the resulting sequence tables. However, if I manually trim the sequence length of the column names and then use collapseNoMismatch(), I get the error message "Mismatch between output sequences and the collapsed sequence table." So I am wondering if there is a better way to go about this, without having to go all the way back to the filterAndTrim() step for each MiSeq run. 

Here's my code, where seqtab is the merged sequence table:

```
seqlengths <- sapply(colnames(seqtab, nchar))
seqtab75.temp <- seqtab[,-which(seqlengths<75)]
colnames(seqtab75.temp) <- substr(colnames(seqtab)[-which(seqlengths<75)], 1, 75)
seqtab75 <- collapseNoMismatch(seqtab75.temp)
```

If it makes any difference, the sequences are ITS1, so I wanted to learn the error rates on the variable length sequences. Now I want to see if trimming and collapsing at different sequence lengths (to 75, 90, 100, 150 base pairs) can retain the spread of community diversity. 

Thanks for any help you provide!

**Comment by ltipton** [comment]
Thank you for the fast response. The code worked great with the following modifications to lines 3 and 4:

`st <- matrix(0L, nrow=nrow(seqtab75.temp), ncol=length(unq75)
rownames(st) <- rownames(seqtab75.temp)`

I had thought that collapseNoMismatch would collapse identical sequences, not thinking about the fact that under normal use, the function would not encounter identical sequences.

**Comment by benjjneb** [comment]
Re-opening and tagging as a bug since `collapseNoMismatch` should handle this case.

**Comment by benjjneb** [comment]
Should be fixed now in ad8b0983e91e547a2e4b47306be265ec302dc415

Also, you can now collapse just identical sequences (for example those that arise from truncating the sequences after making the sequence table) with `collapseNoMismatch(..., identicalOnly=TRUE)`
